macOS/debugger.h

```diff
@@ -196,6 +201,8 @@ friend kern_return_t catch_mach_exception_raise_state_identity(
 private:
   static std::unordered_map<task_t, Debugger*> task_to_debugger_map;
   static std::mutex map_mutex;
+  std::unordered_map<mach_vm_address_t, mach_vm_address_t> share_mem_map;
```

Comment:
These two (shared_mem_map, shared_mem_port) can be combined into

```
struct ShreadMemory {
  mach_vm_address_t local_address;
  mach_vm_address_t remote_address;
  mach_vm_size_t size;
  mach_port_t port;
};

std::list<ShreadMemory> shaded_memory;
```

or similar.